### ISSUE TITLE
Modal window not announcing the new context

### ISSUE THREAD

**Issue by israelcefrin** [pkp/pkp-lib#11259]
## Issue Description
When the user opens a modal from the submission list, the focus is set to the "close" button and the modal title (heading level 1) is not announced. Also, if the user hits the enter again, the modal closes immediately. 

## How to reproduce / Current behavior
Using a screenreader e.g. VoiceOver
1) Log in to an OJS journal
2) Navigate using the TAB key to reach a `View` button
3) Activate the button using `ENTER` key
4) Listen the close element being announced when the modal loads

## What application are you using?
 OJS version 3.5rc2

## Stack used 
- Device: laptop
- OS: MacOS sequoia
- Browser: Edge
- Version: 135.0.3179.54 (Official build) (arm64)

## Testing method
- Method: manual using Keyboard and screenreader VoiceOver

## Action
The modal should load while the browser focus being set to the heading level 1 to announce the title of the modal.

## Additional information
**Focus** 
![Image](https://github.com/user-attachments/assets/3c2d4c0c-908c-4712-8afe-ed58474bc8fa)

**Comment by jardakotesovec** [comment]
@israelcefrin @Tribunal33 
This should be addressed with the changes from https://github.com/pkp/pkp-lib/issues/11258

Can you confirm its solving the issue?

Its available on the jarda.testing instance.

**Comment by israelcefrin** [comment]
Hello @jardakotesovec 

I've tested in jarda.testing instance and the focus is still on the arrow/close button.

**Comment by jardakotesovec** [comment]
Right.. I went through this [one](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/#examples) and that indeed advises to focus on heading for big modals, which should be generally good fit for our side modals.

For dialogs it goes with default behaviour which is selecting with first focusable element.

@israelcefrin Updated on jarda.testing.

**Comment by israelcefrin** [comment]
@jardakotesovec 
I've tested and it works smoothly. 🥇

**Comment by Tribunal33** [comment]
I used Narrator and Edge to check this one and I think it is working correctly.  I'm not sure, however, what the expected requirement is as I am getting some extra info announced.  It immediately announce the  `{author} dialogue` but then it loads and repeats with the `{author} at level 1`.  Then it does seem to go backwards and announce the top level action buttons as `##common.help## graphic, tasks  A`, 

So in full when I select the `View` button from submission list and it opens up the submission I hear this `{Author} dialogue, {Author} at level 1, ##common.help## graphic tasks A`

I see that the focus is on the back arrow < and the announcement is reading the Heading Level 1 - Author?  Is this correct?

**Comment by Tribunal33** [comment]
Ok I think this might be an issue with Narrator.  I swapped over to NVDA and it doesn't announce the top navigation control buttons.  I'm good to close this now.  Maybe we can enhance this in the future but as written this is great!

**Comment by Tribunal33** [comment]
only tested on personal VM so I will re-open this and move to In progress until it is merged to our branches.

**Comment by jardakotesovec** [comment]
Merged to 3.5 and main

@Tribunal33 Feel free to close it, depending if you want to do some additional testing.

**Comment by Tribunal33** [comment]
Works on main.  Announces main level one heading for new context.  Added accessibility test for this as well.
